### PR TITLE
Cmake formatting

### PR DESCRIPTION
**Pull request by mcneish1**
* add cmake-format.json (only non-defaults are tab size, set to 4, and keyword_case, set to "upper")
 * manually change `######...` to `# ----------...`, as it deletes multiple `#`s
 * this only covers ./CMakeLists.txt and the ones in src and test
 * cmake-format converts indented comments to bulleted lists sometimes. I don't think any of that happened here, but it did when I ran on the config directory
   * (those changes aren't in here, I'm still looking through them)

### CODE REVIEW

File: .cmake-format.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "line_width": 80,
+  "dangle_parens": false,
```
Comment (phlptp) [inline .cmake-format.json]: this probably should be true

File: src/fmi_export/CMakeLists.txt [hunk 2]
```diff
@@ -1,113 +1,137 @@
...
+if(WIN32)
+    if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+        install(TARGETS fmiGridDynSharedLib
+                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/win64/ COMPONENT
+                RUNTIME)
+        set(FMILIBRARY_TYPE win64)
+    else("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+        install(TARGETS fmiGridDynSharedLib
+                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/win32/ COMPONENT
+                RUNTIME)
```
Comment (phlptp) [inline src/fmi_export/CMakeLists.txt]: guess we need to rename this component so it isn't thought of as keyword

File: src/gridDynLoader/CMakeLists.txt [hunk 3]
```diff
@@ -1,79 +1,79 @@
...
+            ${PROJECT_BINARY_DIR}/libraryLoader.cpp)
 
-set_target_properties (gridDynLoader PROPERTIES FOLDER libraries)
+set_target_properties(gridDynLoader PROPERTIES FOLDER libraries)
 
-#message("optlibs: ${griddyn_optional_libraries}")
-#message("optheaders: ${griddyn_optional_include_dirs}")
-#message("external_libraries: ${external_library_list}")
-#message("external_lib_dirs: ${external_link_directories}")
+# message("optlibs: ${griddyn_optional_libraries}") message("optheaders:
```
Comment (phlptp) [inline src/gridDynLoader/CMakeLists.txt]: this is odd, and not ideal

File: src/gridDynMain/CMakeLists.txt [hunk 4]
```diff
@@ -28,10 +22,18 @@ if(DIME_ENABLE)
...
+                                                 # copy_if_different..."
+                               "${keyfile}" # <--this is in-file
+                               $<TARGET_FILE_DIR:gridDynMain>) # <--this is out-
+                                                               # file path
 endforeach(keyfile)
 
-INSTALL(TARGETS gridDynMain RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT runtime)
+install(TARGETS gridDynMain
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT
+        RUNTIME)
```
Comment (phlptp) [inline src/gridDynMain/CMakeLists.txt]: runtime should not be capitalized,  should rename the component I guess

File: src/networking/CMakeLists.txt [hunk 5]
```diff
@@ -1,135 +1,122 @@
...
-list(APPEND networking_sources ${zmqLibrary_sources})
-list(APPEND networking_headers ${zmqLibrary_headers})
-list(APPEND networking_sources ${zmqInterface_sources})
-list(APPEND networking_headers ${zmqInterface_headers})
+# LLNS Copyright Start Copyright (c) 2014-2018, Lawrence Livermore National
+# Security This work was performed under the auspices of the U.S. Department of
+# Energy by Lawrence Livermore National Laboratory in part under Contract
+# W-7405-Eng-48 and in part under Contract DE-AC52-07NA27344. Produced at the
+# Lawrence Livermore National Laboratory. All rights reserved. For details, see
+# the LICENSE file. LLNS Copyright End
```
Comment (phlptp) [inline src/networking/CMakeLists.txt]: We may have to reformat the copyright statements to make them work right in the formatter

File: src/gridDynMain/CMakeLists.txt (hunk 4 above)
Comment (mcneish1) [reply src/gridDynMain/CMakeLists.txt]: This is a bug in cmake-format: https://github.com/cheshirekow/cmake_format/issues/66. We can rename it if that's what you want to do.

File: src/gridDynLoader/CMakeLists.txt (hunk 3 above)
Comment (mcneish1) [reply src/gridDynLoader/CMakeLists.txt]: This and the license text reformatting are the same issue. I'm going to see if there's a way to turn off comment line reflow

File: src/fmi_export/CMakeLists.txt (hunk 2 above)
Comment (mcneish1) [reply src/fmi_export/CMakeLists.txt]: same as https://github.com/LLNL/GridDyn/pull/37#pullrequestreview-143301380

File: .cmake-format.json (hunk 1 above)
Comment (mcneish1) [reply .cmake-format.json]: I'm going to have to redo this PR because of the comment reflow, I'll change this over then

**Comment by phlptp** [comment]
Go ahead and rename if that is the simplest solution.




On August 6, 2018 at 10:57:37 AM PDT, Corey McNeish <notifications@github.com> wrote:

@mcneish1 commented on this pull request.

________________________________

In src/gridDynMain/CMakeLists.txt<https://github.com/LLNL/GridDyn/pull/37#discussion_r207978442>:

>  endforeach(keyfile)

-INSTALL(TARGETS gridDynMain RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT runtime)
+install(TARGETS gridDynMain
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT
+        RUNTIME)


This is a bug in cmake-format: cheshirekow/cmake_format#66<https://github.com/cheshirekow/cmake_format/issues/66>. We can rename it if that's what you want to do.

File: src/gridDynMain/CMakeLists.txt (hunk 4 above)
Comment (phlptp) [reply src/gridDynMain/CMakeLists.txt]: Lets go ahead and rename it maybe "binaries"

**Comment by mcneish1** [comment]
The latest 2 commits are a restart, with comment-reflow disabled (https://github.com/cheshirekow/cmake_format/issues/67#issue-348034491), `runtime` renamed to `binaries`, and new config options applied beforehand. ~Going to cut files out of the reformat if the tests are still broken.~

~EDIT: yep, still broken. Seeing if undoing `test/` is enough.~

**Comment by mcneish1** [comment]
Found it: https://github.com/LLNL/GridDyn/pull/37/commits/b8d1a36bfba8342cb55dc0bf965d10f68640f1c3#diff-af3b638bc2a3e6c650974192a53c7291L606

Going to change these manually.

**Comment by phlptp** [comment]
@mcneish1  is this ready to go?

**Comment by mcneish1** [comment]
Yes, I'll make another PR for my other build system changes

**Review by phlptp — APPROVED** [review]

**Pull request merged**